BricksAndHearts.UnitTests/ControllerTests/Landlord/LandlordControllerTests.cs

```diff
@@ -87,4 +87,59 @@ public void EditProfilePage_CalledUsingUserEmail_ReturnsEditProfileViewWithLandl
         // Assert   
         result!.Model.Should().BeOfType<LandlordProfileModel>();
     }
+
+    [Fact]
+    public async void TieUserWithLandlord_WithNonExistentLink_RedirectToInvite()
+    {
+        // Arrange 
+        var landlordUser = CreateLandlordUser();
+        const string inviteLink = "11111";
+        A.CallTo(() => LandlordService.LinkExistingLandlordWithUser(inviteLink,landlordUser))
+            .Returns(ILandlordService.LinkUserWithLandlordResult.ErrorLinkDoesNotExist);
+        MakeUserPrincipalInController(landlordUser, UnderTest);
+
+        // Act
+        var result = await UnderTest.TieUserWithLandlord(inviteLink);
+        
+        // Assert
+        result.Should().BeOfType<RedirectToActionResult>().Which.ActionName.Should().Be("Invite");
+        result.Should().BeOfType<RedirectToActionResult>().Which.RouteValues.Should()
+            .Contain("inviteLink",inviteLink);

```

Comment:
Would be good to check the flash message here as well. Can hopefully just access `UnderTest.TempData["FlashMessage"]` and check it e.g. `.Contains("Invite Link")`

Reply:
yeah sure done

Reply:
Don't need the `?` in `UnderTest.TempData["FlashMessage"].Should().BeNull()` or in the others below. In fact you shouldn't have them because then if it were null the assertion will not run.

`Should` here is an extension method and can accept null.

Reply:
sure done :-)